### PR TITLE
FOUR-19000: Create a NEW VIEW for cases Part List

### PR DESCRIPTION
**Pull request by henryjonathanquispe**
## Related Tickets & Packages
- https://processmaker.atlassian.net/browse/FOUR-19000

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

ci:next

### CODE REVIEW

File: resources/jscomposition/system/table/cell/CaseTitleCell.vue [hunk 1]
```diff
@@ -0,0 +1,46 @@
+<template>
+  <div>
+    <div
+      class="tw-text-nowrap tw-whitespace-nowrap tw-overflow-hidden tw-text-ellipsis"
+      :style="{ width: column.width + 'px' }"
+      v-if="row.case_title_formatted"
```
Comment (fagubla) [inline resources/jscomposition/system/table/cell/CaseTitleCell.vue:6]: v-if must to go first

File: resources/jscomposition/system/table/cell/ParticipantsCell.vue [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    <div
+      class="tw-group"
+      v-for="(participant, index) in participants">
+      <AppTooltip position="bottom">
+        <AppAvatar
+          :key="index"
+          :class="`${participant.color}`"
+          :initials="participant.name[0]"
+          class="tw-cursor-pointer">
+        </AppAvatar>
```
Comment (fagubla) [inline resources/jscomposition/system/table/cell/ParticipantsCell.vue:12]: The platform already have the Component Avatar

File: resources/jscomposition/system/table/cell/ParticipantsCell.vue [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+        <template #content> 
+            <div class="tw-p-2 tw-rounded-lg tw-border tw-border-gray-300 tw-bg-white">
+                {{ participant.name }}
+            </div>
+        </template>
```
Comment (fagubla) [inline resources/jscomposition/system/table/cell/ParticipantsCell.vue:18]: use a correct indentation

File: resources/jscomposition/system/table/cell/StatusCell.vue [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  },
+  ERROR: {
+    color: "red",
+    label: "Error",
+  },
+  IN_PROGRESS: {
+    color: "green",
+    label: "In progress",
+  },
+};
```
Comment (fagubla) [inline resources/jscomposition/system/table/cell/StatusCell.vue:34]: Use these colors primary, warning, danger, success.
In order to keep the compatibility with customizeUI

File: resources/jscomposition/system/table/cell/TruncatedOptionsCell.vue
Comment (fagubla) [inline resources/jscomposition/system/table/cell/TruncatedOptionsCell.vue:1]: use a correct indentation

File: resources/jscomposition/system/table/cell/TruncatedOptionsCell.vue [hunk 5]
```diff
@@ -0,0 +1,97 @@
+<template>
+	<div class="tw-relative ">
+		<div class="tw-text-nowrap tw-whitespace-nowrap tw-overflow-hidden tw-text-ellipsis"
+			:style="{maxWidth: column.width + 'px'}">
+			<a
+				href="#"
+				v-for="(item, index) in row[column.field]"
```
Comment (fagubla) [inline resources/jscomposition/system/table/cell/TruncatedOptionsCell.vue:7]: v-for a :key must go on first

File: resources/jscomposition/system/table/cell/TruncatedOptionsCell.vue [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+				:key="index"
+				class="hover:tw-text-blue-400
+					[&:not(:last-child)::after]:tw-content-['|'] after:tw-text-gray-400/70 after:tw-text-lg"
+				@click.stop.self="onClick($event, item, index)"
+					>
```
Comment (fagubla) [inline resources/jscomposition/system/table/cell/TruncatedOptionsCell.vue:12]: use a correct indentation

File: resources/jscomposition/system/table/cell/TruncatedOptionsCell.vue [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+			default: () => ({}),
+		},
+	},
+	setup(props, { emit }) {
+		const show = ref(false)
+		const optionsModel = ref()
+
+		const onClick = (event, item, index)=>{
+			if(isTruncated(event.currentTarget.parentNode) &&
+				event.currentTarget.getBoundingClientRect().right > event.currentTarget.parentNode.getBoundingClientRect().right){
```
Comment (fagubla) [inline resources/jscomposition/system/table/cell/TruncatedOptionsCell.vue:67]: add a extra space before {

File: resources/jscomposition/base/ui/AppTooltip.vue [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+      const wrapperRect = tooltipWrapper.value.getBoundingClientRect();
+      const tooltipRect = tooltip.value.getBoundingClientRect();
+      const viewportWidth = window.innerWidth;
+      const viewportHeight = window.innerHeight;
+
+      let top = 0;
+      let left = 0;
+
+      // Position calculated dynamicaly
+      switch (props.position) {
```
Comment (fagubla) [inline resources/jscomposition/base/ui/AppTooltip.vue:59]: missing the default option

File: resources/jscomposition/base/ui/AppTooltip.vue [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+
+    // Recalculated position when property visible changes
+    const unwatch = watch(() => props.visible, (newVal) => {
+        if (newVal) {
+          showTooltip();
+        } else {
+          hideTooltip();
+        }
+      }
+    );
```
Comment (fagubla) [inline resources/jscomposition/base/ui/AppTooltip.vue:124]: use a correct indentation

**Review by fagubla — CHANGES_REQUESTED** [review]

File: resources/jscomposition/system/table/cell/StatusCell.vue (hunk 4 above)
Comment (henryjonathanquispe) [reply resources/jscomposition/system/table/cell/StatusCell.vue:34]: The status colors in the new design are not equals that old status colors

File: resources/jscomposition/cases/casesMain/system/table/cell/StatusCell.vue [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+    </span>
+  </div>
+</template>
+<script>
+import { defineComponent, computed } from "vue";
+
+export const statuses = {
+  DRAFT: {
+    color: "red",
+    label: "Draft",
```
Comment (fagubla) [inline resources/jscomposition/cases/casesMain/system/table/cell/StatusCell.vue:16]: the color for draft is warning , could be a yellow

File: resources/jscomposition/system/table/cell/StatusCell.vue (hunk 4 above)
Comment (fagubla) [reply resources/jscomposition/system/table/cell/StatusCell.vue:34]: Draft is no red

**Review by fagubla — CHANGES_REQUESTED** [review]

**Review by fagubla — APPROVED** [review]

**Pull request merged**